### DISCUSSION TITLE
Understanding of --proteins flag

### DISCUSSION THREAD

**Discussion by Rridley7** [oschwengers/bakta#169] [Q&A] [answered]
Hi, 
I wanted to better understand how the `--proteins` flag along with the cds section of bakta works. Are all proteins passed with this flag used as a filter for annotations with those predicted by prodigal? Or how are they otherwise treated?

**Comment by oschwengers** [comment, marked as answer]
Hi and thanks for this question!
The `--proteins` option is *not* used to filter `CDS` but to improve their annotation. If users have a trusted set of proteins along with decent annotations that are not included in the standard database or for which a user has better (more descriptive, more specific) annotations, *then* the `--proteins` option offers a simple mechanism to feed these into the normal functional annotation workflow of Bakta.

So, it does *not* interfere in the gene prediction, regional annotation or filtering, but in the functional annotation of predicted protein-coding genes (CDS). Via the following two simple Fasta header schema, user can provide a lot of information.

Via the short schema, users can provide a gene symbol, a product description and database cross-references. Here, standard alignment thresholds are used, *i.e.* 90% seq identity and 80% mutual query/subject coverage:
```
# short:
>id gene~~~product~~~dbxrefs
MAQ...
```

If required/desired, these alignment thresholds can be adopted via a longer and more expressive schema:
```
# long:
>id min_identity~~~min_query_cov~~~min_subject_cov~~~gene~~~product~~~dbxrefs
MAQ...
```

These are then used as a user provided proteins expert annotation system.

**Reply by Rridley7** [reply]
Very useful, thanks for this response!

**Comment by anjalijain22** [comment]
Hello @oschwengers, thank you for developing and maintaining such a great tool! I have a question regarding the `--proteins` flag. I'm using Bakta with a custom set of ~100 proteins and ran it twice - once with the default sequence identity threshold (90%) and once with 30%. I noticed some unexpected differences between the two runs:

1. At 90% identity, several genes receive different gene and product annotations. For example, in the screenshot below, the same locus is annotated as dmlR, but at 30% identity, it becomes ampR. It seems that even though WP_003160458.1 has a 100% sequence identity match at that locus, Bakta prioritizes the user-provided protein with 30% identity to P24734.3 (custom annotations in red). Is this expected? And is there a way to have Bakta report both annotations - one for the perfect match and one for the lower-identity user protein?

2. How can I retrieve sequence identity metrics for annotations coming from user-defined proteins? The inference.tsv file seems to include these metrics only for Bakta’s internal annotations.

Thanks so much for your help!

<img width="1296" height="230" alt="Screenshot 2025-10-23 at 4 50 01 PM" src="https://github.com/user-attachments/assets/d0f88f27-1c72-466b-851c-9991f41350e2" />

**Reply by anjalijain22** [reply]
Hi @oschwengers, I quickly wanted to follow up on my above question. I would really appreciate any input you may have!

**Reply by oschwengers** [reply]
Hi @anjalijain22 , thanks for your question and reaching out. Let me answer in order:
1. Yes, this is the expected behaviour. Bakta will always prioritize user-provided annotations (and their thresholds), even if those are less specific or result from less strict hits. Otherwise, it would be possible to "overwrite" Bakta internal annotations if they are wrong.
2. Good catch! Your are right. Currently, user-provided inferences are not written to the `*.inference.tsv` file. This is not expected, and I'll write a fix for this. Thank you!

**Reply by anjalijain22** [reply]
Thank you so much for your reply @oschwengers! 
1. Thank you for explaining! I think that makes sense given that Bakta is using the user provided annotations as the ground truth. So I guess the only way to get Bakta to report both annotations - one for the perfect match (with internal annotations) and one for the lower-identity user protein would be to run Bakta twice with default thresholds and the other with 30% PI threshold respectively, is that correct?
2. Awesome, thank you so much for working on a fix for this! I was able to find the metrics in the JSON file after digging around the output files but it would certainly be helpful to have the metrics in the inference.tsv file.
